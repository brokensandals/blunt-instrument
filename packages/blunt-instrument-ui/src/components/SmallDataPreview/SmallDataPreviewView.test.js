import React from 'react';
import renderer from 'react-test-renderer';
import SmallDataPreviewView from './SmallDataPreviewView';
import { Encoder } from 'object-graph-as-json';

class Foo {
  constructor(x) {
    this.x = x;
  }
}

describe('SmallDataPreviewView', () => {
  const encoder = new Encoder();
  const cases = [
    ['boolean false', false],
    ['boolean true', true],
    ['string', 'Hello World! It is so nice to meet a world. Are you a good world?'],
    ['number', 12345678],
    // eslint-disable-next-line no-undef
    ['bigint', encoder.encode(BigInt('1234567890123456789'))],
    ['builtin', { type: 'builtin', name: 'globalThis' }],
    ['unnamed function', encoder.encode(function(){ return 'hi' })],
    ['named function', encoder.encode(function greet(){ return 'hello' })],
    ['short array', encoder.encode([1, 2, 3, 4])],
    ['array with object', encoder.encode([1, 2, { foo: 'bar' }, 3, 4])],
    ['object', encoder.encode({ foo: { bar: 'baz' } })],
    ['object with prototype', encoder.encode(new Foo(10))],
    ['symbol with description', encoder.encode(Symbol('whatevs'))],
    ['symbol without description', encoder.encode(Symbol())],
  ];

  cases.forEach(([name, value]) => {
    test(name, () => {
      const result = renderer.create(<SmallDataPreviewView data={value} />).toJSON();
      expect(result).toMatchSnapshot();
    });
  });

  test('fn-start trev', () => {
    const value = {
      this: { foo: 'bar' },
      arguments: { 0: 'baz', },
      meh: 'baz',
    };
    const result = renderer.create(<SmallDataPreviewView data={encoder.encode(value)} trevType="fn-start" />).toJSON();
    expect(result).toMatchSnapshot();
  });
});
