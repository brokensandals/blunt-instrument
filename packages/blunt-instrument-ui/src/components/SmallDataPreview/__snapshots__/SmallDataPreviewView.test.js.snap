// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallDataPreviewView array with object 1`] = `
Array [
  <span
    className="preview-array-start"
  >
    [
  </span>,
  <span
    className="preview-number preview-array-fixed-element"
  >
    1
  </span>,
  <span
    className="preview-array-comma"
  >
    ,
  </span>,
  <span
    className="preview-number preview-array-fixed-element"
  >
    2
  </span>,
  <span
    className="preview-array-comma"
  >
    ,
  </span>,
  <span
    className="preview-object-start"
  >
    {
  </span>,
  <span
    className="preview-object-key"
  >
    foo
  </span>,
  <span
    className="preview-object-colon"
  >
    :
  </span>,
  <span
    className="preview-string "
  >
    "bar"
  </span>,
  <span
    className="preview-object-end"
  >
    }
  </span>,
  <span
    className="preview-array-comma"
  >
    ,
  </span>,
  <span
    className="preview-number "
  >
    3
  </span>,
  <span
    className="preview-array-comma"
  >
    ,
  </span>,
  <span
    className="preview-number "
  >
    4
  </span>,
  <span
    className="preview-array-end"
  >
    ]
  </span>,
]
`;

exports[`SmallDataPreviewView bigint 1`] = `
<span
  className="preview-bigint "
>
  1234567890123456789
</span>
`;

exports[`SmallDataPreviewView boolean false 1`] = `
<span
  className="preview-boolean "
>
  false
</span>
`;

exports[`SmallDataPreviewView boolean true 1`] = `
<span
  className="preview-boolean "
>
  true
</span>
`;

exports[`SmallDataPreviewView builtin 1`] = `
<span
  className="preview-builtin "
>
  globalThis
</span>
`;

exports[`SmallDataPreviewView fn-start trev 1`] = `
Array [
  <span
    className="preview-object-start"
  >
    {
  </span>,
  <span
    className="preview-object-key"
  >
    meh
  </span>,
  <span
    className="preview-object-colon"
  >
    :
  </span>,
  <span
    className="preview-string "
  >
    "baz"
  </span>,
  <span
    className="preview-object-comma"
  >
    ,
  </span>,
  <span
    className="preview-object-more"
  >
    ...
  </span>,
  <span
    className="preview-object-end"
  >
    }
  </span>,
]
`;

exports[`SmallDataPreviewView named function 1`] = `
<span
  className="preview-function "
>
  greet
</span>
`;

exports[`SmallDataPreviewView number 1`] = `
<span
  className="preview-number "
>
  12345678
</span>
`;

exports[`SmallDataPreviewView object 1`] = `
Array [
  <span
    className="preview-object-start"
  >
    {
  </span>,
  <span
    className="preview-object-key"
  >
    foo
  </span>,
  <span
    className="preview-object-colon"
  >
    :
  </span>,
  <span
    className="preview-object-start"
  >
    {
  </span>,
  <span
    className="preview-object-key"
  >
    bar
  </span>,
  <span
    className="preview-object-colon"
  >
    :
  </span>,
  <span
    className="preview-string "
  >
    "baz"
  </span>,
  <span
    className="preview-object-end"
  >
    }
  </span>,
  <span
    className="preview-object-end"
  >
    }
  </span>,
]
`;

exports[`SmallDataPreviewView object with prototype 1`] = `
Array [
  <span
    className="preview-object-start"
  >
    {
  </span>,
  <span
    className="preview-object-key"
  >
    x
  </span>,
  <span
    className="preview-object-colon"
  >
    :
  </span>,
  <span
    className="preview-number "
  >
    10
  </span>,
  <span
    className="preview-object-comma"
  >
    ,
  </span>,
  <span
    className="preview-object-more"
  >
    ...
  </span>,
  <span
    className="preview-object-end"
  >
    }
  </span>,
]
`;

exports[`SmallDataPreviewView short array 1`] = `
Array [
  <span
    className="preview-array-start"
  >
    [
  </span>,
  <span
    className="preview-number preview-array-fixed-element"
  >
    1
  </span>,
  <span
    className="preview-array-comma"
  >
    ,
  </span>,
  <span
    className="preview-number preview-array-fixed-element"
  >
    2
  </span>,
  <span
    className="preview-array-comma"
  >
    ,
  </span>,
  <span
    className="preview-number preview-array-fixed-element"
  >
    3
  </span>,
  <span
    className="preview-array-comma"
  >
    ,
  </span>,
  <span
    className="preview-number preview-array-fixed-element"
  >
    4
  </span>,
  <span
    className="preview-array-end"
  >
    ]
  </span>,
]
`;

exports[`SmallDataPreviewView string 1`] = `
<span
  className="preview-string "
>
  "Hello World! It is so nice to meet a world. Are you a good world?"
</span>
`;

exports[`SmallDataPreviewView symbol with description 1`] = `
<span
  className="preview-symbol "
>
  ~
  whatevs
</span>
`;

exports[`SmallDataPreviewView symbol without description 1`] = `
<span
  className="preview-symbol "
>
  ~
  14
</span>
`;

exports[`SmallDataPreviewView unnamed function 1`] = `
<span
  className="preview-function "
>
  function () {
    re...
</span>
`;
